tests/ci_tests_basic.yml

```diff
@@ -0,0 +1,83 @@
+---
+- name: empty play to gather facts
+  hosts: all
+- import_playbook: playbooks/rhel7_deps.yml
+  when:
+    - ansible_distribution == "RedHat"
+    - ansible_distribution_major_version == '7'
+- import_playbook: playbooks/centos7_deps.yml
+  when:
+    - ansible_distribution == "CentOS"
+    - ansible_distribution_major_version == '7'
+- name: Run tox tests and nmstatectl show
+  hosts: all
+  vars:
+    - nmstatedir: /tmp/nmstate-git-checkout
+  tasks:
+    # Does not work when specifying multiple hosts:
+    # https://github.com/ansible/ansible/issues/27973
+    # do not run on CentOS/RHEL 6
+    #- meta: end_play
+    #  when: ansible_distribution_major_version == '6'
+    - service:
+        state: started
+        enabled: yes
+        name: NetworkManager
+
+    - shell: 'tar -cvf ../nmstate.tar --exclude "*.pyc" --exclude "__pycache__" --exclude nmstate.tar ../'
+      delegate_to: localhost
+
+    - file:
+        state: directory
+        path: "{{ nmstatedir }}"
+
+    - copy:
+        src: ../nmstate.tar
+        dest: "{{ nmstatedir }}"
+
+    - command: tar xf nmstate.tar
+      args:
+        chdir: "{{ nmstatedir }}"
+
+    - synchronize:
+        src: ../../
+        dest: "{{ nmstatedir }}"
+        recursive: yes
+        delete: yes
+        # tox reqires it to be a .git directory, so cannot exclude .git
+        rsync_opts:
+          - "--exclude=.pyc"
+          - "--exclude=__pycache__"
+      when: false
+
+    - command: git clean -fX
+      args:
+        chdir: "{{ nmstatedir }}"
+
+    - pip:
+        name: tox
+        extra_args: --user
+
+    - command: ~/.local/bin/tox
+      args:
+        chdir: "{{ nmstatedir }}"
+
+    # this needs https://github.com/ansible/ansible/pull/40793 to be merged
```

Comment:
Please add TODO: to the beggining. Is this blocking merging of this patch?

Reply:
No, there is another workaround for the broken pip module in the next task, I added comments as well.